pinot-core/src/main/java/org/apache/pinot/core/accounting/PerQueryCPUMemAccountantFactory.java

```diff
@@ -79,44 +84,45 @@ public static class PerQueryCPUMemResourceUsageAccountant extends Tracing.Defaul
       return thread;
     });
 
-    // number of total threads
-    private final int _numThreads;
     private final PinotConfiguration _config;
-    private final RunnerWorkerThreadOffsetProvider _runnerWorkerThreadOffsetProvider;
 
-    // query_id, task_id per runner/worker thread
-    private final CPUMemThreadLevelAccountingObjects.TaskEntryHolder[] _taskStatus;
+    private final ConcurrentHashMap<Thread, CPUMemThreadLevelAccountingObjects.ThreadEntry> _threadEntriesMap
```

Comment:
(nit) can you add a note that on server this will have both query runner and worker threads ? Similarly for broker.

Reply:
added